prov/rxm/src/rxm_ep.c

```diff
@@ -723,9 +724,13 @@ static ssize_t rxm_ep_send_common(struct fid_ep *ep_fid, const struct iovec *iov
 
 	ret = fi_send(rxm_conn->msg_ep, pkt, pkt_size, tx_buf->hdr.desc, 0, tx_entry);
 	if (ret) {
-		if (ret != -FI_EAGAIN)
+		if ((ret == -FI_EAGAIN) && rxm_ep->progress) {
+			rxm_ep->progress = 0;
```

Comment:
It looks like progress can be a local variable, rather than stored with the EP, which could be racy in a multi-thread environment.